src/lambertw_omega.jl

```diff
@@ -0,0 +1,35 @@
+# lazy-initialized LambertW Omega at 256-bit precision
+const lambertw_Omega_BigFloat256 = Ref{BigFloat}()
+
+# compute BigFloat Omega constant at arbitrary precision
+function compute_lambertw_Omega()
+    # initialize lambertw_Omega_BigFloat256
+    isassigned(lambertw_Omega_BigFloat256) ||
+        (lambertw_Omega_BigFloat256[] = BigFloat("0.5671432904097838729999686622103555497538157871865125081351310792230457930866845666932194", 256))
+    o = BigFloat(lambertw_Omega_BigFloat256[], precision(BigFloat)) # initial value with current precision
+    precision(o) <= 256 && return o
+    # iteratively improve the precision of the constant
+    myeps = eps(BigFloat)
+    for _ in 1:100
```

Comment:
Is this enough for higher precision? It seems a bit dangerous to upper bound the number of iterations.
```suggestion
    while true
```
or alternatively move the stopping criterion here.

Reply:
Wiki says (I've just added the reference) that this is the quadratic method, the number of correct digits is doubled at each iteration. So *100* should be safe for any reasonable precision.
`while true` sounds a bit dangerous.
I can increase it to 1000 and add a warning that the precision was not reached.

Reply:
Maybe just use the stopping criterion, i.e., `while abs(next_o - o) > eps`?

Reply:
I want to avoid dead loop if there's some subtle bug in `BigFloat` implementation or something like that.

Reply:
What bug could cause a dead loop? Generally, we should assume that `BigFloat` does not contain any bugs - and if we notice any, they should be fixed upstream.

BTW just came across the following `while true` loop, I still think it would be a simple and straightforward implementation here as well: https://github.com/JuliaLang/julia/blob/3d11f7db65a3461320542aee3b0f26619c4e65e3/base/irrationals.jl#L54